infra/modules/gcp-oauth-secrets-role/main.tf

```diff
@@ -0,0 +1,14 @@
+resource "google_project_iam_custom_role" "secret_manager_updater" {
```

Comment:
For Jira cloud it is required to have a shared access token. Refresh token can be used once, if a new refresh token is served the prev one will be invalidated. For this, it is needed to build a shared access token with a lock thing.
See the comment on `SecretManager` for GCP for more details, but basically this is the custom role with custom permissions to update/list secrets; otherwise only an admin could do it.

Reply:
also, this isn't oauth in particular; so should name the module differently. It's maybe `gcp-instance-managed-secret` or something?